plugins/annotation/__init__.py

```diff
@@ -0,0 +1,383 @@
+        label="Attribute Type",
+        description="The type of attribute to create.",
+        view=types.View(space=6),
+    )
+    attribute_schema.define_property(
+        "values",
+        types.List(types.String()),
+        label="Values",
+        description="The classes to include in the attribute.",
+        default=[],
+    )
+    return attribute_schema
+
+
+def create_class_schema(ctx):
+    class_schema = types.Object()
+    class_schema.define_property(
+        "classes",
+        types.List(types.String()),
+        label="Classes",
+        required=True,
+        view=types.View(space=6),
+    )
+    class_schema.define_property(
+        "attributes",
+        types.List(create_attribute_schema(ctx)),
+        label="Attributes",
+        required=True,
+        view=types.View(space=6),
+    )
+    return class_schema
+
+
+def create_field_schema(ctx):
+    field_schema = types.Object()
+    field_schema.define_property(
+        "field_name",
+        types.String(),
+        label="Field Name",
+        description="The name of the field.",
+        required=True,
+        view=types.View(space=6),
+    )
+    field_schema.define_property(
+        "type",
+        types.Enum(
+            [
+                "detections",
+                "polygons",
+                "polylines",
+                "points",
+                "text",
+                "scalar",
+                "classifcation",
+                "classifcations",
+            ]
+        ),
+        label="Field Type",
+        description="The type of the field.",
+        view=types.View(space=6),
+    )
+    field_schema.define_property(
+        "classes",
+        types.List(create_class_schema(ctx)),
+        label="Classes",
+        description="The classes to include in the field.",
+        default=[],
+    )
+    return field_schema
+
+
+def define_custom_label_schema(ctx, inputs):
+    inputs.define_property(
+        "label_schema_fields",
+        types.List(create_field_schema(ctx)),
+        label="Label Schema Fields",
+        description="The fields to include in the label schema.",
+        default=[],
+    )
+
+
+def create_anno_schema(ctx):
+    inputs = types.Object()
+    view = types.View(label="Request Annotations")
+    default_view = types.View(space=6)
+    target_choices = types.RadioGroup()
+    target_choices.add_choice(
+        "dataset",
+        label="Dataset",
+        description="Request annotations for the entire dataset",
+    )
+    target_choices.add_choice(
+        "view",
+        label="View",
+        description=f"Request annotations for the current view ({ctx.view.count()} samples)",
+    )
+    target_choice = ctx.params.get("target", None)
+    inputs.enum(
+        "target",
+        target_choices.values(),
+        label="Choose what to annotate",
+        view=target_choices,
+    )
+    inputs.str("annotation_key", label="Annotation Key", required=True)
+    # all backends
```

Comment:
this only lists all configured backends right? If none, maybe it would be helpful to return a message linking to the docs or something

Reply:
Good idea 👍